src/pages/home/ReportScreen.js

```diff
@@ -292,24 +307,51 @@ function ReportScreen({
             firstRenderRef.current = false;
             return;
         }
+
+        const onyxReportID = report.reportID;
+        const prevOnyxReportID = prevReport.reportID;
+        const routeReportID = getReportID(route);
+
+        // Navigate to the Concierge chat if the room was removed from another device (e.g. user leaving a room)
+        if (
+            // non-optimistic case
+            (!prevUserLeavingStatus && userLeavingStatus) ||
+            // optimistic case
+            (prevOnyxReportID && prevOnyxReportID === routeReportID && !onyxReportID && prevReport.statusNum === CONST.REPORT.STATUS.OPEN && report.statusNum === CONST.REPORT.STATUS.CLOSED)
+        ) {
+            Navigation.goBack();
+            Report.navigateToConciergeChat();
+            return;
+        }
+
         // If you already have a report open and are deeplinking to a new report on native,
         // the ReportScreen never actually unmounts and the reportID in the route also doesn't change.
         // Therefore, we need to compare if the existing reportID is the same as the one in the route
         // before deciding that we shouldn't call OpenReport.
-        const onyxReportID = report.reportID;
-        const routeReportID = getReportID(route);
         if (onyxReportID === prevReport.reportID && (!onyxReportID || onyxReportID === routeReportID)) {
             return;
         }
 
         fetchReportIfNeeded();
         ComposerActions.setShouldShowComposeInput(true);
-    }, [route, report, errors, fetchReportIfNeeded, prevReport.reportID]);
+    }, [route, report, errors, fetchReportIfNeeded, prevReport.reportID, prevUserLeavingStatus, userLeavingStatus, prevReport.statusNum]);
+
+    useEffect(() => {
+        // Ensures subscription event succeeds when the report/workspace room is created optimistically.
```

Comment:
@fedirjh Thanks for your feedback. We did it [here](https://github.com/Expensify/App/blob/6cc905388954bdcec3be705b743595f0a3f41a03/src/libs/actions/Report.js#L192-L194) and [here](https://github.com/Expensify/App/blob/6cc905388954bdcec3be705b743595f0a3f41a03/src/libs/actions/Report.js#L242-L244).

The actual issue is this function:
https://github.com/Expensify/App/blob/6cc905388954bdcec3be705b743595f0a3f41a03/src/pages/home/ReportScreen.js#L128-L131

it returned 'null' string instead of an actual `null`.

I'd recommend to change it to:

```
function getReportID(route) {
    const reportID = lodashGet(route, 'params.reportID', null);
    return reportID !== null ? String(reportID) : null;
}
```

@luacmartins, @allroundexperts, considering that this has passed the regression period, the buggy function is not directly related to this PR, and any change would require re-testing. What should we do in this situation? Will there be any adjustments?

Reply:
@fedirjh seems you've resolved it [here](https://github.com/Expensify/App/pull/29590). thank you

Reply:
I think we also need to check onyx report `reportId` availability to avoid https://github.com/Expensify/App/issues/38174